### PR TITLE
Remove InMemory persistence

### PR DESCRIPTION
**Pull request by aleksandr-samila**
Initial version

### CODE REVIEW

File: src/NServiceBus.Core/obsoletes-v8.cs [hunk 1]
```diff
@@ -73,7 +73,7 @@ public static class InMemoryGatewayPersistenceConfigurationExtensions
             Message = "Gateway persistence has been moved to the NServiceBus.Gateway dedicated package.",
             RemoveInVersion = "9.0.0",
             TreatAsErrorFromVersion = "8.0.0")]
-        public static void GatewayDeduplicationCacheSize(this PersistenceExtensions<InMemoryPersistence> persistenceExtensions, int maxSize)
+        public static void GatewayDeduplicationCacheSize(this PersistenceExtensions<LearningPersistence> persistenceExtensions, int maxSize)
```
Comment (andreasohlund) [inline src/NServiceBus.Core/obsoletes-v8.cs:76]: This should be InMemoryPersistence

File: src/NServiceBus.Core.Tests/Unicast/LoadHandlersBehaviorTests.cs [hunk 2]
```diff
@@ -12,11 +14,11 @@ public class LoadHandlersBehaviorTests
         [Test]
         public void Should_throw_when_there_are_no_registered_message_handlers()
         {
-            var behavior = new LoadHandlersConnector(new MessageHandlerRegistry(), new InMemorySynchronizedStorage(), new InMemoryTransactionalSynchronizedStorageAdapter());
+            var behavior = new LoadHandlersConnector(new MessageHandlerRegistry(), new AcceptanceTestingSynchronizedStorage(), new FakeTransactionalSynchronizedStorageAdapter());
```
Comment (andreasohlund) [inline src/NServiceBus.Core.Tests/Unicast/LoadHandlersBehaviorTests.cs:17]:
```suggestion
            var behavior = new LoadHandlersConnector(new MessageHandlerRegistry(), new FakeSynchronizedStorage(), new FakeTransactionalSynchronizedStorageAdapter());
```

File: src/NServiceBus.Core.Tests/Unicast/LoadHandlersBehaviorTests.cs [hunk 3]
```diff
@@ -1,4 +1,6 @@
-﻿namespace NServiceBus.Unicast.Tests
+﻿using NServiceBus.Core.Tests.Fakes;
```
Comment (andreasohlund) [inline src/NServiceBus.Core.Tests/Unicast/LoadHandlersBehaviorTests.cs:1]: Move this inside the namespace

File: src/NServiceBus.Core.Tests/Unicast/LoadHandlersBehaviorTests.cs [hunk 4]
```diff
@@ -24,4 +24,4 @@ public void Should_throw_when_there_are_no_registered_message_handlers()
             Assert.That(async () => await behavior.Invoke(context, c => Task.CompletedTask), Throws.InvalidOperationException);
         }
     }
-}
\ No newline at end of file
+}
```
[the remark this answers is not in the document]
Comment (aleksandr-samila) [reply src/NServiceBus.Core.Tests/Unicast/LoadHandlersBehaviorTests.cs:26]: No, I didn't do anything

[the remark this answers is not in the document]
Comment (andreasohlund) [reply src/NServiceBus.Core.Tests/Unicast/LoadHandlersBehaviorTests.cs:26]: @bording you being the git master when it comes to line-ending issues, do U have any ideas why this happens?

**Review by andreasohlund — CHANGES_REQUESTED** [review]

File: src/NServiceBus.Core.Tests/Unicast/LoadHandlersBehaviorTests.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (ramonsmits) [reply src/NServiceBus.Core.Tests/Unicast/LoadHandlersBehaviorTests.cs:26]: @andreasohlund Its VS code formatting, pretty sure, this isn't line-endings as then all lines would be affected. This is EOF CR/LF.

**Review by ramonsmits — APPROVED** [review]

File: src/NServiceBus.Core.Tests/Unicast/LoadHandlersBehaviorTests.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (bording) [reply src/NServiceBus.Core.Tests/Unicast/LoadHandlersBehaviorTests.cs:26]: As @ramonsmits mentioned, this isn't a line ending problem. Git likes to make it clear if there are newline characters on the final line of the file or not.

In this case, there was no newline at the end of the file, and now there is.

[the remark this answers is not in the document]
Comment (ramonsmits) [reply src/NServiceBus.Core.Tests/Unicast/LoadHandlersBehaviorTests.cs]: @aleksandr-samila do check if you have this checkbox set:

image

File: src/NServiceBus.Core.Tests/Fakes/FakeTimeoutPersistence.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
+﻿namespace NServiceBus.Core.Tests.Fakes
+{
+    using NServiceBus.Features;
+    using NServiceBus.Timeout.Core;
+    using Microsoft.Extensions.DependencyInjection;
+
+    class FakeTimeoutPersistence : Feature
```
Comment (andreasohlund) [inline src/NServiceBus.Core.Tests/Fakes/FakeTimeoutPersistence.cs:7]: I think this isn't used and can be deleted?

**Review by andreasohlund — APPROVED** [review]

File: src/NServiceBus.Core.Tests/Unicast/LoadHandlersBehaviorTests.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (aleksandr-samila) [reply src/NServiceBus.Core.Tests/Unicast/LoadHandlersBehaviorTests.cs]: It's unchecked
image

**Pull request merged**